pages/administration/productions/[productionSlug]/permissions.vue

```diff
@@ -20,120 +21,144 @@
   </AdminPage>
 </template>
 
-<script lang="ts" setup>
+<script>
 import Swal from 'sweetalert2';
 
 import PermissionsAssigner from '@/components/admin/permissions/PermissionsAssigner.vue';
-import { getValidationErrors } from '~~/utils/api';
+import { getValidationErrors, performMutation } from '~~/utils/api';
 import { loadingSwal, successToast } from '~~/utils/alerts';
 import AllErrorsDisplay from '@/components/ui/AllErrorsDisplay.vue';
 import {
   AdminProductionPermissionsDocument,
-  type AdminProductionPermissionsQuery,
-  type AdminProductionPermissionsQueryVariables,
-  useProductionPermissionsMutationsMutation
+  ProductionPermissionsMutationsDocument
 } from '~~/graphql/codegen/operations';
-import Errors from '~~/classes/Errors';
 
-const productionSlug = useRoute().params.productionSlug;
-
-if (typeof productionSlug !== 'string')
-  throw createSafeError({ message: 'Only one slug can be passed' });
-
-const { data, refresh } = await useAsyncQuery<AdminProductionPermissionsQuery>({
-  query: AdminProductionPermissionsDocument,
-  variables: {
-    slug: productionSlug
-  } satisfies AdminProductionPermissionsQueryVariables,
-  cache: false
-});
-
-const production = computed(() => data.value?.production);
-
-if (!production.value) {
-  throw createSafeError({
-    statusCode: 404,
-    message: 'This production does not exist'
-  });
-}
-if (production?.value?.assignedUsers === null) {
-  throw createSafeError({
-    statusCode: 401,
-    message: 'You do not have permission to alter permissions'
-  });
-}
-
-const errors = ref<Errors | undefined>(undefined);
-
-async function savePermissions() {
-  const changedUsers = production.value?.assignedUsers
-    ? production.value.assignedUsers.filter((user: any) => user?.modified)
-    : [];
-
-  const mutations: (boolean | Promise<boolean>)[] = changedUsers.map(
-    (assignedUser) => {
-      if (!assignedUser?.user) return false;
-      return setUserPermissions(
-        assignedUser.user.email,
-        (assignedUser.assignedPermissions?.filter(
-          (perm) => perm !== null
-        ) as string[]) ?? []
-      );
+export default defineNuxtComponent({
+  components: {
+    PermissionsAssigner,
+    AllErrorsDisplay
+  },
+  async asyncData() {
+    // Execute query
+    const { data } = await useDefaultApolloClient().query({
+      query: AdminProductionPermissionsDocument,
+      variables: {
+        slug: useRoute().params.productionSlug
+      },
+      fetchPolicy: 'no-cache'
+    });
+
+    const production = data.production;
+
+    if (!production) {
+      throw createSafeError({
+        statusCode: 404,
+        message: 'This production does not exist'
+      });
+    }
+    if (production.assignedUsers === null) {
```

Comment:
How does this query check for perms? Does `production.assignedUsers` only return data if the requesting user has perms?

Reply:
Yes, this is handled on the API side, which will only return data that a user has permission to view. It may be, therefore, that a user has permission to view a production, but not permission to edit assigned users, in which case they will recieve a partial response from the API wherein `production.assignedUsers` is null